app/views/partials/_received_requests.html.erb

```diff
@@ -0,0 +1,23 @@
+<div class="rr">
+  <% unless @received_requests.empty? %>
+    <h3>Received Requests</h3>
+    <ul class="received_requests">
+      <% @received_requests.each do |user| %>
+        <% unless current_user.friend?(user) %>
+          <li>
+            <%= user.name %>
+            <span class="profile-link">
+              <%= link_to 'See Profile',  user_path(user), class: 'profile-link' %>
+                <%= link_to accept_request_user_friendships_path(user), data: {confirm: "Are you sure about this?" } do %>
+                  <button>Accept</button>
+                <% end %>
+                <%= link_to decline_request_user_friendships_path(user), data: {confirm: "Are you sure about this?" } do %>
+                  <button>Decline</button>
+                <% end %>
+            </span>
+          </li>
+        <% end %>
```

Comment:
- [ ] create a helper method that contains the unless statement calling a partial containing the markup within the unless statement and then replace lines 6 to 19 with the helper method call instead of having the complexity straight in the view.

Reply:
Hi, can i get you to help me out

Reply:
Actually, thanks for pointing me in this direction as I learnt a new technique of doing things... Can I contact you anytime if I need help with technical questions?